src/actions/breakpoints/syncBreakpoint.js

```diff
@@ -63,38 +84,19 @@ export async function syncClientBreakpoint(
   const existingClient = client.getBreakpointByLocation(generatedLocation);
 
   if (existingClient) {
-    const newGeneratedLocation = await getGeneratedLocation(
-      getState(),
-      source,
-      location,
-      sourceMaps
-    );
-
-    if (locationMoved(generatedLocation, newGeneratedLocation)) {
-      await client.removeBreakpoint(generatedLocation);
-      await client.setBreakpoint(
-        newGeneratedLocation,
-        pendingBreakpoint.condition,
-        sourceMaps.isOriginalId(sourceId)
-      );
+    if (!locationMoved(generatedLocation, scopeGeneratedLocation)) {
+      assertBreakpoint(breakpoint);
+      return { breakpoint, previousLocation: location };
     }
 
-    const breakpoint = {
-      ...pendingBreakpoint,
-      id: makeLocationId(location),
-      generatedLocation: newGeneratedLocation,
-      location: location
-    };
-
-    assertBreakpoint(breakpoint);
-    return { breakpoint, previousLocation: location };
+    await client.removeBreakpoint(generatedLocation);
```

Comment:
don't we need do `client.setBreakpoint` to create a new one?

Reply:
its called a little bit lower (the two cases are now more like one case)